src/jsifier.js

```diff
@@ -45,7 +47,8 @@ function JSify(data, functionsOnly) {
     // else. This lets us not hold any strings in memory, we simply print
     // things out as they are ready.
 
-    var shellParts = read(shellFile).split('{{BODY}}');
+    var thirdParty = thirdPartyFiles.map(function(f) { return read(f) }).join('\n');
+    var shellParts = read(shellFile).replace('{{THIRD_PARTY}}', thirdParty).split('{{BODY}}');
```

Comment:
> This seems to add a third party library sodium to all generated builds? That is awkward for build sizes ... gate the generated code on `SINGLE_FILE` option to not impact builds which don't utilize `SINGLE_FILE`

Yep, as mentioned above, that was exactly how I wanted to handle this in another commit. It's only 3 KB, which isn't much compared to the size of typical emscripten-generated code, but there's still no reason to include it when it isn't used.

> Could https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding be used instead?

As mentioned, I have simple implementations working that basically do this (the base64 one is a little different conceptually because because it's converting to bytes, not a string):

```
function base64ToBytes (s) {
    var decoded = atob(s);
    var bytes = new Uint8Array(decoded.length);
    for (i = 0 ; i < decoded.length ; ++i) {
        bytes[i] = decoded.charCodeAt(i);
    }
    return bytes;
}

function bytesToString (bytes) {
    if (typeof TextDecoder !== 'undefined') {
        return new TextDecoder('utf-8', {fatal: true}).decode(bytes);
    }
    var s = '';
    for (var i = 0 ; i < bytes.length ; ++i) {
        s += String.fromCharCode(bytes[i]);
    }
    return s;
}
```

I just don't know offhand how well these handle corner cases. Maybe there's a test suite out there that we could use.

> plus also it will trigger a legal review for a number of partner companies, which we would prefer to avoid

There's always my earlier proposed solution of asking the libsodium maintainers to provide `from_base64` and `to_string` as MIT, or we could go further and ask them to fully donate them to emscripten.

(The base64 implementation as I understand is nothing special aside from being constant time, which doesn't matter here, but I believe their string encoding/decoding has been extended to support relevant edge cases like very large byte arrays.)